src/main/java/dev/galacticraft/mod/content/block/special/fluidpipe/GlassFluidPipeBlock.java

```diff
@@ -80,11 +81,16 @@ public void setPlacedBy(Level level, BlockPos blockPos, BlockState blockState, @
                     livingEntity.setItemInHand(interactionHand, copy);
                 }
             }
+
+            // Regular Stuff
+            var changed = false;
             for (var direction : Constant.Misc.DIRECTIONS) {
-                var otherBlockEntity = level.getBlockEntity(blockPos.relative(direction));
-                glassPipe.getConnections()[direction.ordinal()] = (otherBlockEntity instanceof Pipe pipe && pipe.canConnect(direction.getOpposite())) || FluidUtil.canAccessFluid(level, blockPos.relative(direction), direction);
+                changed |= glassPipe.getConnections()[direction.ordinal()] = glassPipe.canConnect(direction) && FluidUtil.canAccessFluid(level, blockPos.relative(direction), direction);
+            }
+            if (changed) {
+                glassPipe.setChanged();
             }
-            level.updateNeighborsAt(blockPos, blockState.getBlock());
+            level.sendBlockUpdated(blockPos, blockState, blockState, Block.UPDATE_IMMEDIATE);
```

Comment:
Could this be moved into the `if (changed)` block? (Do still we need to update the block if no connections were made?)

Reply:
Yeah that makes sense. Done!

Reply:
Also I made this change to wires as well